Tests/SentryTests/TestClient.swift

```diff
@@ -1,18 +1,17 @@
 import Foundation
 
 class TestClient: Client {
-    let sentryFileManager: SentryFileManager
-    let queue = DispatchQueue(label: "TestClient", attributes: .concurrent)
-
-    override init?(options: Options) {
-        sentryFileManager = try! SentryFileManager(options: options, andCurrentDateProvider: TestCurrentDateProvider())
-        super.init(options: options, permissionsObserver: TestSentryPermissionsObserver())
+    override init(options: Options) {
```

Comment:
This init doesn't have to be optional any more because the downstream init no longer has to worry about creating its own file manager, which is what caused the possible error state leading to nil being returned. That helped simplify a lot of the optionality logic around the tests, where all the TestClients were IUOs, etc.